server/coordinator/procedure/procedure.go

```diff
@@ -0,0 +1,45 @@
+// Copyright 2022 CeresDB Project Authors. Licensed under Apache-2.0.
+
+package procedure
+
+import "context"
+
+type State string
+
+const (
+	StateRunning   = "running"
+	StateFinished  = "finished"
+	StateFailed    = "failed"
+	StateCancelled = "cancelled"
+)
+
+type procedureType = uint
+
+const (
+	switchLeader procedureType = iota
+	mergeShard
+)
+
+// Procedure is used to describe how to execute a set of operations from the scheduler, e.g. SwitchLeaderProcedure, MergeShardProcedure.
+type Procedure interface {
+	// ID of the procedure.
+	ID() uint64
+
+	// Type of the procedure.
+	Type() procedureType
```

Comment:
As the return value of an exported interface should be public too, so it should be `ProcedureType`.